clients/pkg/promtail/targets/docker/targetmanager.go

```diff
@@ -73,14 +73,15 @@ func NewTargetManager(
 				_, ok := tm.groups[syncerKey]
 				if !ok {
 					tm.groups[syncerKey] = &targetGroup{
-						metrics:       metrics,
-						logger:        logger,
-						positions:     positions,
-						targets:       make(map[string]*Target),
-						entryHandler:  pipeline.Wrap(pushClient),
-						defaultLabels: model.LabelSet{},
-						relabelConfig: cfg.RelabelConfigs,
-						host:          sdConfig.Host,
+						metrics:          metrics,
+						logger:           logger,
+						positions:        positions,
+						targets:          make(map[string]*Target),
+						entryHandler:     pipeline.Wrap(pushClient),
+						defaultLabels:    model.LabelSet{},
+						relabelConfig:    cfg.RelabelConfigs,
+						host:             sdConfig.Host,
+						httpClientConfig: sdConfig.HTTPClientConfig,
```

Comment:
Maybe a bit out of the scope of this PR but, looking at line 72:
```
syncerKey := fmt.Sprintf("%s/%s:%d", cfg.JobName, sdConfig.Host, sdConfig.Port)
```

Looks like we have a `sdConfig.Port` we don't use anywhere else, and we actually expect the user to set any port number in the host itself.

It would be good to use the configured `sdConfig.Port` unless the port is specified in the host itself.

Reply:
I need to look where and how the port should be used. I think this should be done in another PR.

Reply:
Doing it on a separate PR sounds good to me as long as we leave a TODO somewhere.